browser/extensions/manifest_v2/brave_extensions_manifest_v2_migrator.cc

```diff
@@ -0,0 +1,425 @@
+          .AppendASCII(*webstore_extension_id)
+          .Append(extensions::kLocalExtensionSettingsDirectoryName)
+          .AppendASCII(*webstore_extension_id);
+  return !base::IsDirectoryEmpty(backup) ? backup : base::FilePath();
+}
+
+base::FileEnumerator GetIndexedSettings(
+    const extensions::ExtensionId& extension_id,
+    const base::FilePath& profile_dir) {
+  const auto pattern = ASCIIToPathStringType(
+      base::StrCat({"chrome-extension_", extension_id, "_*indexeddb*"}));
+  return base::FileEnumerator(profile_dir.Append(kIndexedDBDir), false,
+                              base::FileEnumerator::DIRECTORIES, pattern);
+}
+
+base::FileEnumerator GetIndexedSettingsForImport(
+    const extensions::ExtensionId& brave_hosted_extension_id,
+    const base::FilePath& profile_dir) {
+  CHECK(extensions_mv2::IsKnownBraveHostedExtension(brave_hosted_extension_id));
+  const auto webstore_extension_id =
+      extensions_mv2::GetWebStoreHostedExtensionId(brave_hosted_extension_id);
+  CHECK(webstore_extension_id);
+
+  const auto pattern = ASCIIToPathStringType(base::StrCat(
+      {"chrome-extension_", *webstore_extension_id, "_*indexeddb*"}));
+  return base::FileEnumerator(profile_dir.Append(kExtensionMV2BackupDir)
+                                  .AppendASCII(*webstore_extension_id)
+                                  .Append(kIndexedDBDir),
+                              false, base::FileEnumerator::DIRECTORIES,
+                              pattern);
+}
+
+bool IsBackupAvailableFor(
+    const extensions::ExtensionId& brave_hosted_extension_id,
+    const base::FilePath& profile_dir) {
+  CHECK(extensions_mv2::IsKnownBraveHostedExtension(brave_hosted_extension_id));
+
+  const auto webstore_extension_id =
+      extensions_mv2::GetWebStoreHostedExtensionId(brave_hosted_extension_id);
+  if (!webstore_extension_id) {
+    return false;
+  }
+
+  if (!base::PathExists(profile_dir.Append(kExtensionMV2BackupDir)
+                            .AppendASCII(*webstore_extension_id)
+                            .AppendASCII("version"))) {
+    return false;
+  }
+
+  if (!GetLocalSettingsForImport(brave_hosted_extension_id, profile_dir)
+           .empty()) {
+    return true;
+  }
+
+  auto indexed =
+      GetIndexedSettingsForImport(brave_hosted_extension_id, profile_dir);
+  if (!indexed.Next().empty()) {
+    return true;
+  }
+  return false;
+}
+
+base::Version GetBackupVersion(
+    const extensions::ExtensionId& webstore_extension_id,
+    const base::FilePath& profile_dir) {
+  const auto version_path = profile_dir.Append(kExtensionMV2BackupDir)
+                                .AppendASCII(webstore_extension_id)
+                                .AppendASCII("version");
+  if (std::string content;
+      base::ReadFileToStringWithMaxSize(version_path, &content, 256)) {
+    return base::Version(content);
+  }
+  return base::Version();
+}
+
+base::Version BackupExtensionSettingsOnFileThread(
```

Comment:
What if we make a universal method to copy/move settings?
I.e. `MoveExtensionSettings(source_dir, source_id, target_dir, target_id, delete_source)`?
And then just call it twice (for backup and for import)?